app/views/unattended/provisioning_templates/provision/preseed_default.erb

```diff
@@ -125,11 +125,12 @@ d-i apt-setup/local<%= repos %>/key string http://keyserver.ubuntu.com/pks/looku
 <% end -%>
 <% end -%>
 
-<% if host_param('kt_activation_keys') && @host.operatingsystem.name == 'Debian' -%>
-# If we are using Katello for content management, then do not add updates and
-# security repos during the initial installation. Prevents us from defaulting to:
-# d-i apt-setup/services-select multiselect updates,security
+<% if host_param('kt_activation_keys') -%>
+# If we are using Katello for content management, then we do not want to use
+# upstream mirrors prior to Katello registration.
+d-i apt-setup/use_mirror boolean false
 d-i apt-setup/services-select multiselect
+d-i apt-setup/security_host string localhost
```

Comment:
shouldn't those two be not needed once we have `use_mirror` set to false?

Reply:
From a quick look at the `apt-setup` source package, `use_mirror` is only used in `generators/50mirror` and `generators/50mirror.ubuntu`.

`generators/91security` doesn't offer that particular flexibility, and mandatorily sets an entry based on the `apt-setup/security_host` parameter.

Reply:
If **disabling** security support entirely is desired (I wasn't sure what to think about the pointing to `localhost`…), better tweak `apt-setup/services-select` which is defined in Debian with the following:

```
Template: apt-setup/services-select
Type: multiselect
Choices-C: security, updates, backports
Default: security, updates
```

Reply:
Yeah, I was about to say, after I pinged you, I found https://salsa.debian.org/installer-team/apt-setup/-/blob/master/generators/91security#L8 which indicates that if security isn't in the service list, it won't be setup.

Reply:
And yes, disabling is desired in this case, as Foreman/Katello is providing the mirrors locally.

Reply:
In my experience in most cases the 'd-i apt-setup/services-select multiselect' line was sufficient to achieve the desired offline-installation.
You could also add a "none" behind it besides leaving it empty, as far as I know - but the effect should be the same.

Reply:
If I have understood everything correctly there are at least some cases where all of these settings make sense, and they should not harm those Katello cases where they are not absolutely necessary?

=> i.e. I should leave the line as is?

Reply:
I think it has finally clicked for me, how to interpret the `91security` line you referenced. The lack of `security` in `d-i apt-setup/services-select multiselect` will already prevent that entire file from doing anything making the `d-i apt-setup/security_host string localhost` redundant.

I have removed the line.